### ISSUE TITLE
Missing packages and other problems when setting up the environment on AWS

### ISSUE THREAD

**Issue by hanxLi** [NASA-IMPACT/hls-foundation-os#42]
Following the instruction provided, it seems that when building the environment on AWS there are several problems when running the fine-tune process for multi temporal crop classification:

1. `ModuleNotFoundError: No module named 'tifffile'`: This can be fixed by a simple `pip install tifffile`
2. `TypeError: FormatCode() got an unexpected keyword argument 'verify'` : This seems to be a existing issue with mmcv, can be bypassed by running `pip install yapf==0.40.1`
3. When setting the `gpu_ids = range(0, 4)` for running it on more than 1 gpu, the instructed command `mim train mmsegmentation configs/multi_temporal_crop_classification.py`, by default it uses a single gpu. Only when using additional command `CUDA_VISIBLE_DEVICES=0,1,2,3` and `--gpus=4` the model is able to run on multiple gpus
4. The training process works just fine but when the model do eval epochs, similar issue with https://github.com/NASA-IMPACT/hls-foundation-os/issues/41#issue-2029944968 appears.

**Comment by CarlosGomes98** [comment]
Hi @hanxLi,

Thanks for raising these points! I've pushed code that addresses issues 1 and 2, which as you mentioned where easy fixes (thanks to your pointers).

Issue 3 should also be fixed with the new push. MMSeg requires a shape of [1, h, w] for training labels but [h, w] for validation. The code should now be more robust to this invariant of if the segmentation labels are stored as [1, h, w] or [h, w].

3. multi-gpu training is possible using mim by adding the `--gpus <num_gpus>` flag, as well as `--launcher pytorch`. In fact, the gpu_ids in the config file can me ommited, and I have removed them to avoid confusion.

Besides the removal of `gpu_id` from the config file for multi-gpu training, there should not be any other changes needed.

Will wait for your confirmation before closing this.
